cmd/client-fs.go

```diff
@@ -978,12 +978,17 @@ func (f *fsClient) Stat(isIncomplete, isFetchMeta bool) (content *clientContent,
 	content.Metadata = map[string]string{
 		"Content-Type": guessURLContentType(f.PathURL.Path),
 	}
-	path := f.PathURL.String()
-	metaData, pErr := getAllXattrs(path)
-	if pErr != nil {
-		return content, nil
+	if isFetchMeta {
+		path := f.PathURL.String()
+		metaData, pErr := getAllXattrs(path)
+		if pErr != nil {
```

Comment:
why are we returning `nil` error if `getAllXattrs` fails?

Reply:
> why are we returning nil error if getAllXattrs fails?

yeah, a comment is worthy to be added here.

Reply:
extended attributes are optional and to be returned in the metadata if it is available.That's why the error is not cascaded up. @krisis, mc cp never copied metadata other than Content-Type from the source object. While adding mc stat  command, a regression was introduced because the xattr was appended to metadata without checking if isFetchMeta is set to true. Hence the fix.

Reply:
@poornas, it would help if we added this explanation in comments.

Reply:
@krisis, done.